### DISCUSSION TITLE
Support for the missing platforms in the NuGet package

### DISCUSSION THREAD

**Discussion by samuel-lucas6** [jedisct1/libsodium#1366] [Ideas]
People have expressed interest in support for Android and iOS among the .NET bindings:

https://github.com/samuel-lucas6/Geralt/issues/7
https://github.com/ektrah/nsec/issues/73
https://github.com/ektrah/libsodium-core/issues/101
https://github.com/ektrah/libsodium-core/issues/80
https://github.com/ektrah/libsodium-core/pull/89
https://github.com/jedisct1/libsodium/issues/1122
https://github.com/jedisct1/libsodium/pull/1332

There's also interest in support for Windows ARM64:

https://github.com/jedisct1/libsodium/discussions/1360
https://github.com/jedisct1/libsodium/discussions/1219
https://github.com/jedisct1/libsodium/discussions/1104

Would it be possible to add the missing platforms to the NuGet package? How difficult is this/what assistance is required?

I unfortunately haven't done anything with these platforms. If anyone has/is able to help, these users/us maintainers would be extremely grateful for your help.

**Comment by jedisct1** [comment]
ARM64 support is easy to add, but I don't know how to test it 🤷

Libraries for all Apple platforms are built with the `dist-build/apple-xcframework.sh` script, but I don't know what changes are required to use an xcframework in a NuGet package.

**Reply by MartyIX** [reply]
> Libraries for all Apple platforms are built with the `dist-build/apple-xcframework.sh` script, but I don't know what changes are required to use an xcframework in a NuGet package.

@rolfbjarne might know.

In #1238, I updated https://github.com/jedisct1/libsodium/blob/master/packaging/dotnet-core/libsodium.pkgproj, I guess that's needed.

**Reply by rolfbjarne** [reply]
> I don't know what changes are required to use an xcframework in a NuGet package.

There are two main ways:

1. Use a binding project, and reference the .xcframework in a NativeReference item. Then when doing `dotnet pack` the build will put the .xcframework in the right location.
2. Add the .xcframework (or the correct .framework for the tfm in question) in each the runtimes/<tfm> folder. In this case you can also zip up the .xcframework (so you have an .xcframework.zip file) - which is necessary to preserve symlinks if the .xcframework contains them.

**Reply by MartyIX** [reply]
@rolfbjarne Your second options looks like it just involves adding a line in https://github.com/jedisct1/libsodium/blob/74781400dc765e07492970ac973f7eb48cb83460/packaging/dotnet-core/libsodium.pkgproj#L28-L37, is that correct?

**Reply by rolfbjarne** [reply]
You'd need one line for each runtime identifier. If you want to support all the Apple ones: `ios-arm64`, `iossimulator-arm64`, `iossimulator-x64`, `tvos-arm64`, `tvossimulator-arm64`, `tvossimulator-x64`, `maccatalyst-arm64` and `maccatalyst-x64`.

Note that the Mac Catalyst content inside the .xcframework will likely have symlinks, so I'd recommend creating an .xcframework.zip and use that instead of the directory itself.

**Comment by paulverbeke** [comment]
Anyone know how to add support for the Android platform to the NuGet package ?
